pints/tests/test_diagnostics.py

```diff
@@ -79,36 +79,99 @@ def test_within(self):
 
-    def test_reorder(self):
-        # Tests that reorder function reshapes correctly
+    def test_rhat(self):
+        # Tests that rhat works
 
-        test = np.random.normal(loc=0, scale=1, size=(4, 10, 3))
-        y = np.array(pints._diagnostics.reorder(0, test))
-        self.assertEqual(y.shape, (8, 5))
+        # Tests Rhat computation for one parameter, chains.shape=(2, 4)
+        chains = np.array([[1.0, 1.1, 1.4, 1.3],
+                          [1.0, 2.0, 3.0, 4.0]])
+        self.assertAlmostEqual(
+            pints._diagnostics.rhat(chains), 2.3303847470550716, 6)
+
+        # Test Rhat computation for two parameters, chains.shape=(3, 4, 2)
+        chains = np.array([
+            [
+                [-1.10580535, 2.26589882],
+                [0.35604827, 1.03523364],
+                [-1.62581126, 0.47308597],
+                [1.03999619, 0.58203464]
+            ],
+            [
+                [-1.04755457, -2.28410098],
+                [0.17577692, -0.79433186],
+                [-0.07979098, -1.87816551],
+                [-1.39836319, 0.95119085]
+            ],
+            [
+                [-1.1182588, -0.34647435],
+                [1.36928142, -1.4079284],
+                [0.92272047, -1.49997615],
+                [0.89531238, 0.63207977]
+            ]])
+
+        y = pints._diagnostics.rhat(chains)
+        d = np.array(y) - np.array([0.84735944450487122, 1.1712652416950846])
+        self.assertLess(np.linalg.norm(d), 0.01)
 
-    def test_reorder_all(self):
-        # Tests that reorder_all function reshapes correctly
+    def test_bad_rhat_inputs(self):
+        # Tests whether exceptions are thrown should the input to rhat not be
+        # valid
+
+        # Pass chain of dimension 1
+        chains = np.empty(shape=1)
+        message = (
+            'Dimension of chains is 1. '
+            + 'Method computes R^hat for one '
+            'or multiple parameters and therefore only accepts 2 or 3 '
+            'dimensional arrays.')
+        self.assertRaisesRegex(
+            ValueError, message[0], pints.rhat, chains)
+
+        # Pass chain of dimension 4
+        chains = np.empty(shape=(1, 1, 1, 1))
+        message = (
+            'Dimension of chains is 4. '
+            + 'Method computes R^hat for one '
+            'or multiple parameters and therefore only accepts 2 or 3 '
+            'dimensional arrays.')
+        self.assertRaisesRegex(
+            ValueError, message[0], pints.rhat, chains)
 
-        test = np.random.normal(loc=0, scale=1, size=(4, 10, 3))
-        y = np.array(pints._diagnostics.reorder_all_params(test))
-        self.assertEqual(y.shape, (3, 8, 5))
+        # Pass bad warm-up arguments
+        chains = np.empty(shape=(2, 4))
 
-    def test_rhat(self):
-        # Tests that rhat works
+        # warm-up greater than 100%
+        warm_up = 1.1
+        message = (
+            'warum_up is set to 1.1. warm_up only takes values in [0,1].')
```

Comment:
''warm-up''

Reply:
I referring here to the input variable `warm_p`. I would therefore leave it like that. But I can add some `.` marks if that make sit clearer.

Reply:
My comment was just about spelling warm_up rather than warum_up

Reply:
Ahh I see :D